### PR TITLE
common: fix crypto memory

### PR DESCRIPTION
**Pull request by outscale-nts**
Signed-off-by: Tarrade Nicolas <nicolas.tarrade@outscale.com>

### CODE REVIEW

File: api/common/crypto.cc [hunk 1]
```diff
@@ -99,7 +99,7 @@ bool EncAes256CbcSha512(const std::string &key,
     // Build last block content with padding
     memcpy(last_block, clear_last_block, clear.length() % AES_BLOCK_SIZE);
     memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length,
-           AES_BLOCK_SIZE);
+           AES_BLOCK_SIZE - clear.length() % AES_BLOCK_SIZE);
```
Comment (outscale-jju) [inline api/common/crypto.cc]: you already know pad_length so I would say
`memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length, pad_length);`
Check this in tests [crypto.cc unit tests](https://github.com/outscale-nts/butterfly/blob/f6f803bd83cd5b13d56fd630fb58e7ac76f55201/api/common/crypto.cc#L229)

File: api/common/crypto.cc [hunk 2]
```diff
@@ -98,8 +98,7 @@ bool EncAes256CbcSha512(const std::string &key,
 
     // Build last block content with padding
     memcpy(last_block, clear_last_block, clear.length() % AES_BLOCK_SIZE);
-    memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length,
-           AES_BLOCK_SIZE);
+    memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length, pad_length);
```
Comment (outscale-mgo) [inline api/common/crypto.cc]: I'm not sure about this one:
first the 2nd argument should be a character, here it's a length (which happen to he a character too, but a little random)
2: I think the memset shoud set what hasn't been set by memcpy first, not overrite data
so maybe 3rd argument should be (pad_length - clear.length() % AES_BLOCK_SIZE)

File: CMakeLists.txt [hunk 3]
```diff
@@ -43,13 +43,16 @@ set(PG_LIBRARIES
     -lz -lnuma)
 set(PG_INCLUDE_DIR ${3RDPARTY_DIR}/packetgraph/include)
 
-set(CMAKE_C_FLAGS "-g -O3 -march=core-avx-i -mtune=core-avx-i -fmessage-length=0  -Werror -Wall -Wextra -Wwrite-strings -Winit-self -Wcast-align -Wpointer-arith -Wstrict-aliasing -Wformat=2 -Wmissing-declarations -Wmissing-include-dirs -Wno-unused-parameter -Wuninitialized -Wold-style-definition -Wstrict-prototypes -Wmissing-prototypes -L${DPDK_INSTALL_DIR}/build/lib")
+set(CMAKE_C_FLAGS "-g3 -O3 -march=core-avx-i -mtune=core-avx-i -fmessage-length=0  -Werror -Wall -Wextra -Wwrite-strings -Winit-self -Wcast-align -Wpointer-arith -Wstrict-aliasing -Wformat=2 -Wmissing-declarations -Wmissing-include-dirs -Wno-unused-parameter -Wuninitialized -Wold-style-definition -Wstrict-prototypes -Wmissing-prototypes -L${DPDK_INSTALL_DIR}/build/lib")
```
Comment (outscale-mgo) [inline CMakeLists.txt]: why not, but why in this commit ?
do we really need g3 ?

File: api/common/crypto.cc [hunk 4]
```diff
@@ -169,7 +169,7 @@ bool DecAes256CbcSha512(const std::string &key,
     EVP_CIPHER_CTX *aes = NULL;
     EVP_MD_CTX *sha;
 
-    std::unique_ptr<uint8_t> clear_result(new uint8_t[blocks_len]);
+    std::unique_ptr<uint8_t[]> clear_result(new uint8_t[blocks_len]);
```
Comment (outscale-mgo) [inline api/common/crypto.cc]: I would  prefer that kind of fix in another commit (like the previous one)

**Pull request merged**